internal/controller/stages/event_handlers.go

```diff
@@ -478,6 +513,11 @@ func (p *phaseChangedAnalysisRunHandler[T]) Update(
 			)
 		}
 		for _, stage := range stages.Items {
+			// If the Stage is a control flow Stage, there is no need to reconcile it.
+			if stage.IsControlFlow() {
+				continue
```

Comment:
Previously, this would always result in a reconciliation — even if there wasn't any actual interest.